docs/strategies/circuit-breaker.md

```diff
@@ -99,3 +99,229 @@ await manualControl.CloseAsync();
+  - If a future maintainer of code changes the `circuitBreaker`'s `BreakDuration` then (s)he might forget to change sleep duration as well
+
+✅ DO
+
+Use `Context` to pass information between strategies
+
+<!-- snippet: circuit-breaker-pattern-1 -->
+```cs
+var circuitBreaker = new ResiliencePipelineBuilder()
+    .AddCircuitBreaker(new()
+    {
+        ShouldHandle = new PredicateBuilder().Handle<HttpRequestException>(),
+        BreakDuration = TimeSpan.FromSeconds(5),
+        OnOpened = static args =>
+        {
+            args.Context.Properties.Set(SleepDurationKey, args.BreakDuration);
+            return ValueTask.CompletedTask;
+        },
+        OnClosed = args =>
+        {
+            args.Context.Properties.Set(SleepDurationKey, null);
+            return ValueTask.CompletedTask;
+        }
+    })
+    .Build();
+
+var retry = new ResiliencePipelineBuilder()
+    .AddRetry(new()
+    {
+        ShouldHandle = new PredicateBuilder()
+            .Handle<HttpRequestException>()
+            .Handle<BrokenCircuitException>(),
+        DelayGenerator = static args =>
+        {
+            _ = args.Context.Properties.TryGetValue(SleepDurationKey, out var delay);
+            delay ??= TimeSpan.FromSeconds(1);
+            return ValueTask.FromResult(delay);
+        }
+    })
+    .Build();
+```
+<!-- endSnippet -->
+
+**Reasoning**:
+- With this approach the two strategies are less coupled
+  - They both use the context and the `sleepDurationKey` components
+- The Circuit Breaker shares the `BreakDuration` via the context whenever it breaks
+  - When it transitions back to Closed then it "revokes" the sharring
+- The Retry do not have any circuit breaker specific logic
+  - It dynamically fetches the sleep duration whithout knowing anything about the circuit breaker
+  - Also if `BreakDuration` needs to be adjusted then it can be done in a single place
+
+### 2 - Using different duration for breaks
+
+In case of Retry you can specify dynamically the sleep duration via the `DelayGenerator`.
+In case of Circuit Breaker the `BreakDuration` is considered constant (can't be changed between breaks).
+
+❌ DON'T
+
+Use `Task.Delay` inside `OnOpened`
+
+<!-- snippet: circuit-breaker-anti-pattern-2 -->
+```cs
+static IEnumerable<TimeSpan> GetSleepDuration()
+{
+    for (int i = 1; i < 10; i++)
+    {
+        yield return TimeSpan.FromSeconds(i);
+    }
+}
+
+var sleepDurationProvider = GetSleepDuration().GetEnumerator();
+sleepDurationProvider.MoveNext();
+
+var circuitBreaker = new ResiliencePipelineBuilder()
+    .AddCircuitBreaker(new()
+    {
+        ShouldHandle = new PredicateBuilder().Handle<HttpRequestException>(),
+        BreakDuration = TimeSpan.FromSeconds(0.5),
+        OnOpened = async args =>
+        {
+            await Task.Delay(sleepDurationProvider.Current);
+            sleepDurationProvider.MoveNext();
+        }
+
+    })
+    .Build();
+```
+<!-- endSnippet -->
+
+**Reasoning**:
```

Comment:
## Reasoning

- The minimum break duration value is half a second. This implies that each sleep lasts for `sleepDurationProvider.Current` plus an additional half second.
- One might assume that setting the `BreakDuration` to `sleepDurationProvider.Current` would address this, but it doesn't. This is because the `BreakDuration` is established only once and isn't reassessed during each break.